### ISSUE TITLE
net->copytrainedlayer failed.

### ISSUE THREAD

**Issue by yytzjgsu** [TheLegendAli/DeepLab-Context#29]
I have successfully compiled the code and trained a model using my own data.Then I wrote a c++ script to test a single image .But unfortunately, net->copytrainedlayer(model) seems not work ,weights of layers are still 0.Can you help me or just give me some clues ?
ps, I highly doubt that if this function can not work well,the training process may be also incorrect.

I0413 11:50:28.412251  9144 net.cpp:170] relu1_2 needs backward computation.
I0413 11:50:28.412251  9144 net.cpp:170] conv1_2 needs backward computation.
I0413 11:50:28.413254  9144 net.cpp:170] relu1_1 needs backward computation.
I0413 11:50:28.413254  9144 net.cpp:170] conv1_1 needs backward computation.
I0413 11:50:28.414253  9144 net.cpp:172] data does not need backward computation.
I0413 11:50:28.414253  9144 net.cpp:208] This network produces output accuracy
I0413 11:50:28.414253  9144 net.cpp:467] Collecting Learning Rate and Weight Decay.
I0413 11:50:28.416255  9144 net.cpp:219] Network initialization done.
I0413 11:50:28.416255  9144 net.cpp:220] Memory required for data: 2445378832
I0413 11:50:28.417255  9144 solver.cpp:41] Solver scaffolding done.
I0413 11:50:28.417255  9144 caffe.cpp:118] Finetuning from voc12/model/vgg128_noup/init.caffemodel
I0413 11:50:28.417255  9144 net.cpp:740] Target layer conv1_1 not initialized.
I0413 11:50:28.418267  9144 net.cpp:740] Target layer conv1_2 not initialized.
I0413 11:50:28.418267  9144 net.cpp:740] Target layer conv2_1 not initialized.
I0413 11:50:28.419257  9144 net.cpp:740] Target layer conv2_2 not initialized.
I0413 11:50:28.419257  9144 net.cpp:740] Target layer conv3_1 not initialized.
I0413 11:50:28.420267  9144 net.cpp:740] Target layer conv3_2 not initialized.
I0413 11:50:28.420267  9144 net.cpp:740] Target layer conv3_3 not initialized.
Could not create logging file: File exists
COULD NOT CREATE A LOGGINGFILE 20170413-115028.4828!I0413 11:50:28.421259  9144 net.cpp:740] Target layer conv4_1 not initialized.
I0413 11:50:28.422260  9144 net.cpp:740] Target layer conv4_2 not initialized.
I0413 11:50:28.422260  9144 net.cpp:740] Target layer conv4_3 not initialized.
I0413 11:50:28.422260  9144 net.cpp:740] Target layer conv5_1 not initialized.
I0413 11:50:28.423260  9144 net.cpp:740] Target layer conv5_2 not initialized.
I0413 11:50:28.423260  9144 net.cpp:740] Target layer conv5_3 not initialized.
I0413 11:50:28.424262  9144 net.cpp:740] Target layer fc6 not initialized.
I0413 11:50:28.424262  9144 net.cpp:740] Target layer fc7 not initialized.
I0413 11:50:28.425261  9144 net.cpp:740] Target layer fc8_voc12 not initialized.
I0413 11:50:28.426262  9144 solver.cpp:160] Solving vgg128_noup
I0413 11:50:28.426262  9144 solver.cpp:161] Learning Rate Policy: step
I0413 11:50:29.027125  9144 solver.cpp:209] Iteration 0, loss = 3.04452
I0413 11:50:35.966346  9144 solver.cpp:224]     Train net output #0: accuracy = 0
I0413 11:50:35.967347  9144 solver.cpp:224]     Train net output #1: accuracy = 0
I0413 11:50:35.967347  9144 solver.cpp:224]     Train net output #2: accuracy = 0.571429

**Comment by yytzjgsu** [comment]
The model I trained may not work.But function "copyfromtrained" should still copy weights from the trained model when my deploy.ptototxt has exactly the same layer names with train.prototxt.

**Comment by danea32** [comment]
hi,i wonder how can you  successfully compiled the code and trained a model using your own data,could you please tell me how to do that ,thank you very much!
